src/controller/CHIPDeviceController.cpp

```diff
@@ -685,6 +713,7 @@ void DeviceCommissioner::RendezvousCleanup(CHIP_ERROR status)
 {
     if (mRendezvousSession != nullptr)
     {
+        mNextKeyId = mRendezvousSession->GetNextKeyId();
```

Comment:
Do you think it would be worth extracting:
```
    if (mRendezvousSession != nullptr)
    {
        mNextKeyId = mRendezvousSession->GetNextKeyId();
        chip::Platform::Delete(mRendezvousSession);
        mRendezvousSession = nullptr;
    }
```
into a separate function `FreeRendezvousSession` not to repeat the sequence several times?